docs/usage/STATE_SYNC.md

```diff
@@ -66,27 +66,40 @@ reference should return its `key`.
+the method that transforms(resolves by querying and caching key values for Ids) and maps from a `State` to `StateDraft` using cache in order to make them ready for reference resolution by the sync, for example: 
 
 ````java
-// Build a StateQuery for fetching states from a source CTP project with all the needed references expanded for the sync
-final StateQuery stateQueryWithReferenceExpanded = StateReferenceResolutionUtils.buildStateQuery();
+// Build an StateQuery for fetching shopping lists from a source CTP project without any references expanded for the sync:
+final StateQuery stateQueryWithReferences = StateReferenceResolutionUtils.buildStateQuery();
 
 // Query all states (NOTE this is just for example, please adjust your logic)
 final List<State> states =
     CtpQueryUtils
-        .queryAll(sphereClient, stateQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, stateQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the state, 
+Initialize [`StateReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/states/service/StateReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the transformStateReferences method with the `states` parameter as shown below:
 
-// Mapping from State to StateDraft with considering reference resolution.
-final List<StateDraft> stateDrafts = StateReferenceResolutionUtils.mapToStateDrafts(states);
+````java
+// Fetch(Id to key values for references) into the cache and map from State to StateDraft using cache with considering reference resolution.
+final List<StateDraft> stateDrafts = StateReferenceTransformService.transformStateReferences(states);
 ````
+
+The cache here is used for a better performance. 
+Instead of expanding the references in the query for state resource. `StateReferenceTransformService` will execute a query to fetch key values for the Ids(nonCached) and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Comment:
```suggestion
Instead of expanding the references in the query for state resource. `StateReferenceTransformService` will execute a query to fetch key-id pairs and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```